### PR TITLE
Fix a problem with pattern matching on inf-clojure-repl-type

### PR DESCRIPTION
**Pull request by arichiardi**
It was never matching on the "default" branch with symbols, we switch to
strings.

---

I break it, I fix it :smile:

### CODE REVIEW

File: inf-clojure.el [hunk 1]
```diff
@@ -577,7 +577,7 @@ The prefix argument SWITCH-TO-REPL controls whether to switch to REPL after the
 If you are using REPL types, it will pickup the most approapriate
 `inf-clojure-var-doc-form` variant."
   (pcase inf-clojure-repl-type
-    (lumo inf-clojure-var-doc-form-lumo)
```
Comment (bbatsov) [inline inf-clojure.el]: You simply meant write ``lumo` here.

Comment (arichiardi) [inline inf-clojure.el]: I tried that backtick in `ielm` and it did not work, it was always going to the else branch. Also in `cider` I have never seen using symbols so I settled for strings.

Comment (bbatsov) [inline inf-clojure.el]: I'm 100% that this is the correct syntax for `pcase` symbols and it's generally better to use symbols over strings (strings are not interned in Elisp). If cider uses only strings in such cases it's just accidental.

Comment (arichiardi) [inline inf-clojure.el]: So one backtick in front ? I will try again

**Pull request merged**